internal/handlers/common/projection.go

```diff
@@ -234,3 +258,154 @@ func filterFieldArrayElemMatch(k1 string, doc, conditions *types.Document, docVa
 	}
 	return
 }
+
+// filterFieldArraySlice implements $slice projection query.
+func filterFieldArraySlice(docValue *types.Array, projectionValue any) (*types.Array, error) {
+	switch projectionValue.(type) {
+	// TODO what do we do with int64?
+	case int32, int64, float64:
+		var n int
+		switch pr := projectionValue.(type) {
+		case float64:
+			if math.IsNaN(pr) {
+			} else if math.IsInf(pr, -1) || n < math.MinInt {
+				n = math.MinInt
+			} else if math.IsInf(pr, +1) || n > math.MaxInt {
+				n = math.MaxInt
+			} else {
+				n = int(pr)
+			}
+		case int64:
```

Comment:
Is the question of what to do with `in64` still actual? How it is sounds?

Reply:
No, I believe it's not relevant anymore
I made a stupid mistake there and worked with `int32` instead of `int`